raiden-ts/src/migration/3.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { AddressZero, Zero, HashZero } from 'ethers/constants';
+import findKey from 'lodash/findKey';
+
+import { SignatureZero } from '../constants';
+
+const emptyBP = {
+  chainId: Zero,
+  tokenNetworkAddress: AddressZero,
+  channelId: Zero,
+  nonce: Zero,
+  transferredAmount: Zero,
+  lockedAmount: Zero,
+  locksroot: HashZero,
+  additionalHash: HashZero,
+  signature: SignatureZero,
+};
+
+function migrateChannelEnd(channel: any, endsAddr: any) {
```

Comment:
I don't seem to remember but do we backup the old state before the migration? what happens if for some unforseen reason the migration fails?

Reply:
We don't backup because it's an in-memory object. If it fails, Raiden initialization fails and throws without touching storage. After this batch of migrations are performed successfully, we yet proceed to do a full in-depth validation with `RaidenState.decode`, and only then this migrated state is used and overwrites the original data on local storage.